src/controllers/contactController.js

```diff
@@ -0,0 +1,103 @@
+// Controllers are responsible for handling HTTP requests, processing business logic, and rendering responses. In your application
+
+import { Contact } from "../models/contactModel.js";
+import Joi from "joi";
+
+export const getAll = async (req, res) => {
+  let contacts;
+  try {
+    contacts = await Contact.find({}).sort({ createdAt: "desc" });
+
+    return res.json({ contacts });
+  } catch (error) {
+    return res.status(500).json({ error: error.message });
+  }
+};
+
+export const getById = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const reqContact = await Contact.findOne({ _id: id });
+    if (!reqContact) return res.sendStatus(404);
+    return res.json(reqContact);
+  } catch (error) {
+    return res.status(500).json({ error: error.message });
+  }
+};
+
+export const createContact = async (req, res) => {
+  try {
+    const { name, email, phone } = req.body;
+    if (!name || (!email && !phone))
+      return res
+        .status(400)
+        .json({ error: "Required fields: name and phone or email!" });
+
+    await Joi.object({
+      name: Joi.string().required(),
+      email: Joi.string().email(),
+      phone: Joi.string(),
+    }).validateAsync(req.body);
+
+    const createdContact = await Contact.create({ name, email, phone });
+
+    return res.status(201).json(createdContact);
+  } catch (error) {
+    return res.status(400).json({ error: error.message });
+  }
+};
+
+export const updateById = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const updatedContactFields = req.body;
+
+    await Joi.object({
+      name: Joi.string(),
+      email: Joi.string().email(),
+      phone: Joi.string(),
+    }).validateAsync(updatedContactFields);
```

Comment:
To jest metoda put, co się stanie jeżeli nie będzie phone? 
Przy put jeśli nie ma phone przesłanego to powinno go usunąć, zmodyfikować bazę danych podobnie inne rekordy.